darwin/cli_functions.py

```diff
@@ -520,7 +561,9 @@ def _config():
     return Config(Path.home() / ".darwin" / "config.yaml")
 
 
-def _load_client(team: Optional[str] = None, offline: bool = False, maybe_guest: bool = False):
+def _load_client(
+    team: Optional[str] = None, offline: bool = False, maybe_guest: bool = False, dataset_slug: Optional[str] = None
+):
```

Comment:
If both `team` and `dataset_slug` aren't set, we'll end up having issues.

Also, `dataset_slug` is in the form `<team_slug>/<dataset_slug>`, should we call it `dataset_identifier` instead?
(If so, `team = DatasetIdentifier.parse(dataset_identifier).team_slug` works for me)